app/src/main/java/net/gsantner/markor/widget/TodoWidgetRemoteViewsFactory.java

```diff
@@ -0,0 +1,89 @@
+package net.gsantner.markor.widget;
+
+import android.content.Context;
+import android.content.Intent;
+import android.widget.RemoteViews;
+import android.widget.RemoteViewsService;
+
+import net.gsantner.markor.ApplicationObject;
+import net.gsantner.markor.R;
+import net.gsantner.markor.format.todotxt.TodoTxtTask;
+import net.gsantner.markor.model.AppSettings;
+import net.gsantner.markor.model.Document;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TodoWidgetRemoteViewsFactory implements RemoteViewsService.RemoteViewsFactory {
+
+    private final Context _context;
+    private final AppSettings _appSettings;
+    private final Document _document;
+    private final List<TodoTxtTask> _tasks;
+
+    public TodoWidgetRemoteViewsFactory(Context context, Intent intent) {
+        _context = context;
+        _appSettings = ApplicationObject.settings();
+        _document = new Document(_appSettings.getTodoFile());
+        _tasks = new ArrayList<>();
+    }
+
+    @Override
+    public void onCreate() {
+        onDataSetChanged();
+    }
+
+    @Override
+    public void onDataSetChanged() {
+        _tasks.clear();
+        final String content = _document.loadContent(_context);
+        if (content == null) {
+            return;
+        }
+        List<TodoTxtTask> tasks = TodoTxtTask.getAllTasks(content);
+        _tasks.addAll(tasks);
+    }
+
+    @Override
+    public void onDestroy() {
+        _tasks.clear();
+    }
+
+    @Override
+    public int getCount() {
+        return _tasks.size();
+    }
+
+    @Override
+    public RemoteViews getViewAt(int position) {
+        RemoteViews views = new RemoteViews(_context.getPackageName(), R.layout.todo_widget_list_item);
+        views.setTextViewText(R.id.todo_widget_item_text, _tasks.get(position).getDescription());
```

Comment:
It would be really good to also highlight this.

Can do this pretty easily. See getSttHighlighter in MarkorDialogFactory.

Reply:
Plaintext/no highlight would be enough for the first version. Especially if you anyway think about further changes

Reply:
>Plaintext/no highlight would be enough for the first version.
Especially if you anyway think about further changes

Do you mean, that I should merge this PR and make further changes in another?
I also plan on making another widget for regular notes which would take some time.

Reply:
My plan would be to get a initial version that works well enough / not makes crashes. Prefered over full featured and half stuff bare works 😄

Reply:
Well... It works well on my machine 😅
I cannot find other ways to improve it beyond what I mentioned. Maybe you might find something?